src/nbl/video/utilities/IUtilities.cpp

```diff
@@ -4,40 +4,60 @@
 
 namespace nbl::video
 {
-void IUtilities::updateImageViaStagingBuffer(
-    IGPUCommandBuffer* cmdbuf, IGPUFence* fence, IGPUQueue* queue,
+IGPUQueue::SSubmitInfo IUtilities::updateImageViaStagingBuffer(
     asset::ICPUBuffer const* srcBuffer, asset::E_FORMAT srcFormat, video::IGPUImage* dstImage, asset::IImage::E_LAYOUT dstImageLayout, const core::SRange<const asset::IImage::SBufferCopy>& regions,
-    uint32_t& waitSemaphoreCount, IGPUSemaphore*const * &semaphoresToWaitBeforeOverwrite, const asset::E_PIPELINE_STAGE_FLAGS* &stagesToWaitForPerSemaphore)
+    IGPUQueue::SSubmitInfo intendedNextSubmit, IGPUQueue* submissionQueue, IGPUFence* submissionFence)
 {
+    if(!intendedNextSubmit.isValid())
+    {
+        // TODO: log error
+        assert(false);
+        return intendedNextSubmit;
+    }
+
+    bool noCommandBufferInNextSubmit = (intendedNextSubmit.commandBufferCount == 0);
+    if (noCommandBufferInNextSubmit)
```

Comment:
I could base the detection on whether the last commandbuffer is resettable, and still recording and one time submit

If not then end the last commandbuffer (if still recording) and create a new one like here, plus copy all the old commandbuffers pointers to a unique_ptr new array + the new commandbuffer